finite_state_machine/state_machine.py

```diff
@@ -72,12 +47,37 @@ def transition(source, target, conditions=None, on_error=None):
             raise ValueError("on_error needs to be a bool, int or string")
 
     def transition_decorator(func):
-        func.__fsm = TransitionMeta(func.__name__)
-        if isinstance(source, (list, tuple, set)):
-            for item in source:
-                func.__fsm.add_transition(item, target, on_error, conditions)
+        mems = inspect.getmembers(func)
+        state_machine_instance = [
+            mem[1]["StateMachine"] for mem in mems if mem[0] == "__globals__"
+        ][0]
```

Comment:
Saving instance of StateMachine to later add "__fsm" attribute. It turned out func was type "function" not "method".

Reply:
this could be replced by `state_machine_instance = func.__self__`

Reply:
When I tried this it would not work because `func` is of type function not a method. So I had to dive into the __globals__ referenced for the function type here: https://docs.python.org/3/library/inspect.html. Perhaps I am missing something.

Reply:
Ahhh.... doing a quick google search, doesn't seem like it's possible without inspect. 🤦 

I think a possible workaround might be to use a class as a function decorator. Don't worry about it. This works, let's just go with it. If this library has some magic, that's okay. It makes the code which uses this library cleaner which is really what we want.